### PR TITLE
Add Imprint form

### PR DESCRIPTION
**Pull request by AminAlAit**
### Short description

Creation of Imprint form. Adjusted the code from the page form, page template and page view to suit that of the imprint.

### Proposed changes

- Inside _cms.view.imprints.imprint_view.py_, `page_id` and `new_page` variables were replaced by `imprint_id` and `new_imprint` with no noticeable effect.  

### Resolved issues

- Imprint tab can be clicked from the menu

Fixes: #39

### CODE REVIEW

File: src/cms/forms/imprints/imprint_translation_form.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            self.cleaned_data,
+            self.changed_data,
+        )
+
+        # pop kwarg to make sure the super class does not get this param
+        page = kwargs.pop("page", None)
+        user = kwargs.pop("user", None)
+
+        kwargs["commit"] = False  # Don't save yet. We just want the object.
+        page_translation = super().save(*args, **kwargs)
```
Comment (svenseeberg) [inline src/cms/forms/imprints/imprint_translation_form.py:72]: I think we should rename `page_translation` to `imprint_translation`. Makes sense, doesn't it?

File: src/cms/templates/imprints/imprint_form.html [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+            {% if can_edit_page %}
+                <input type="submit" name="submit_draft" class="bg-gray-500 hover:bg-gray-600 cursor-pointer text-white font-bold py-3 px-4 rounded mr-2" value="{% trans 'Save as draft' %}" />
+            {% endif %}
+            {% has_perm 'cms.publish_page' request.user page as can_publish_page %}
+            {% if can_publish_page %}
+                <input type="submit" name="submit_public" class="cursor-pointer bg-blue-500 hover:bg-blue-600 text-white font-bold py-3 px-4 rounded mr-2" value="{% trans 'Publish' %}" />
+            {% else %}
+                <input type="submit" name="submit_review" class="cursor-pointer bg-blue-500 hover:bg-green-600 text-white font-bold py-3 px-4 rounded mr-2" value="{% trans 'Submit for review' %}" />
+            {% endif %}
+            <button type="button" onclick="short_url_to_clipboard('s/p/{{ page_translation_form.instance.id }}');" class="cursor-pointer bg-blue-500 hover:bg-blue-600 text-white font-bold py-3 px-4 rounded">
```
Comment (svenseeberg) [inline src/cms/templates/imprints/imprint_form.html:78]: Not sure if we need a short URL at all for the imprint. But if we have it, we need to change the path. I'd say `i` for imprint? -> `s/i/IMPRINT_TRANSLATION_ID`.

**Comment by svenseeberg** [comment]
The overall concept seems valid. Thank you for the contribution. Currently, lots of code is doubled. It would be cool to have some abstract classes and maybe even templates, so that the code can be shared with the page translations. @timoludwig do you agree?

**Review by timoludwig — CHANGES_REQUESTED** [review]
Thanks for your work. :+1: 

Please have a look at the [Page](https://integreat.github.io/integreat-cms/autogenerated/cms.models.pages.html#module-cms.models.pages.page) and [ImprintPage](https://integreat.github.io/integreat-cms/autogenerated/cms.models.pages.html#module-cms.models.pages.imprint_page) models and understand, that both have the same base class [AbstractBasePage](https://integreat.github.io/integreat-cms/autogenerated/cms.models.pages.html#module-cms.models.pages.abstract_base_page), but are different models with slightly different attributes etc.
There are multiple places where you just copied+pasted code for the wrong model. Copy+paste is a good start here, because the models are very similar and it doesn't make sense to re-invent the wheel, but since they're not the same, you have to understand the difference between these models and adapt the files you copied.
This includes e.g.:
- Please make sure to use the correct models (`ImprintPage` instead of `Page` etc)
- Use the correct permissions everywhere (`manage_imprint` instead of `edit_page` etc)
- Please remove any fields which are only relevant for the Page model
- Rename all variables

I tried to make suggestions, but probably didn't find all occurrences of wrong model usages etc, so please have a look again at the code you have copied + pasted and try to understand what it's doing.

Apart from that: Please log into [CircleCI](https://app.circleci.com/pipelines/github/Integreat/integreat-cms?branch=develop-amin), grant it access to your GitHub-account and restart the build to fix the "unauthorized" error

**Comment by timoludwig** [comment]
> The overall concept seems valid. Thank you for the contribution. Currently, lots of code is doubled. It would be cool to have some abstract classes and maybe even templates, so that the code can be shared with the page translations. @timoludwig do you agree?

@svenseeberg I have noticed a lot of potential for improvement in our views/forms/templates code (not only page/imprint) and will probably make a major refactoring in the next couple of weeks. But I think it would ok to merge this PR and refactor afterwards.

**Review by AminAlAit — COMMENTED** [review]
.

**Review by timoludwig — APPROVED** [review]
@AminAlAit I rebased your branch onto the current develop branch and fixed all problems I noticed in the views. I think this is ready to merge now :tada:

**Pull request merged**